### PR TITLE
Search string visibility 2

### PR DESCRIPTION
**Pull request by gongchoong**

### CODE REVIEW

**Comment by gongchoong** [comment]
@p0deje 
App crashes when you open Preferences - Appearance.
In AppearanceSettingsViewController.swift,
line 35 
private var imageHeightFormatter: NumberFormatter!

**Comment by p0deje** [comment]
@gongchoong I think the XIB is outdated, I see that you changed all the text fields back to sliders (https://github.com/p0deje/Maccy/pull/728/files#diff-4802f8bfa12187d4de5899afa761392c9f721446c23558dec3164971eb547992R164), which is few versions old. I would recommend you to review the diff to revert unrelated changes in XIB file. Otherwise, you can just revert it completely and -re-add NSPopUpButton again, then re-bind it to controller

File: Maccy/Extensions/UserDefaults+Configuration.swift [hunk 1]
```diff
@@ -232,4 +234,9 @@ extension UserDefaults {
     get { bool(forKey: Keys.suppressClearAlert) }
     set { set(newValue, forKey: Keys.suppressClearAlert) }
   }
+    
+  @objc dynamic public var searchStringHighlight: String {
```
Comment (p0deje) [inline Maccy/Extensions/UserDefaults+Configuration.swift:238]: You don't need `@objc dynamic` here.

File: Maccy/Menu/HistoryMenuItem/HistoryMenuItem.swift [hunk 2]
```diff
@@ -17,13 +17,26 @@ class HistoryMenuItem: NSMenuItem {
...
-  private let highlightFont: NSFont = {
+  private let highlightBoldFont: NSFont = {
     if #available(macOS 11, *) {
       return NSFont.boldSystemFont(ofSize: 13)
     } else {
       return NSFont.boldSystemFont(ofSize: 14)
     }
   }()
+    
+  private let highlightItalicFont: NSFont = {
```
Comment (p0deje) [inline Maccy/Menu/HistoryMenuItem/HistoryMenuItem.swift:28]: I would expect this not to be bold - just italic.

File: Maccy/Menu/HistoryMenuItem/HistoryMenuItem.swift [hunk 3]
```diff
@@ -128,7 +141,14 @@ class HistoryMenuItem: NSMenuItem {
       let highlightRange = NSRange(location: range.lowerBound, length: rangeLength)
 
       if Range(highlightRange, in: title) != nil {
-        attributedTitle.addAttribute(.font, value: highlightFont, range: highlightRange)
+          switch UserDefaults.standard.searchStringHighlight {
+          case "italic":
+            attributedTitle.addAttribute(.font, value: highlightItalicFont, range: highlightRange)
+          case "underline":
+              attributedTitle.addAttributes([.underlineStyle: NSUnderlineStyle.single.rawValue, .font: highlightBoldFont], range: highlightRange)
```
Comment (p0deje) [inline Maccy/Menu/HistoryMenuItem/HistoryMenuItem.swift:148]: 1. I would expect this not to be bold, just underlined.
2. Please fix indentation.

File: Maccy/Settings/AppearanceSettingsViewController.swift [hunk 4]
```diff
@@ -344,5 +357,16 @@ class AppearanceSettingsViewController: NSViewController, SettingsPane {
...
   }
+    
+  private func populateSearchStringHighlight() {
+    switch UserDefaults.standard.searchStringHighlight {
+    case "italic":
+      searchStringHighlightButton.selectItem(withTag: 1)
+    case "underline":
+      searchStringHighlightButton.selectItem(withTag: 2)
+    default:
+        searchStringHighlightButton.selectItem(withTag: 0)
```
Comment (p0deje) [inline Maccy/Settings/AppearanceSettingsViewController.swift:368]: Please fix indentation.

File: Maccy/Settings/Base.lproj/AppearanceSettingsViewController.xib [hunk 5]
```diff
@@ -327,9 +330,38 @@ Default: 1500.</string>
...
+                        </gridCell>
+                        <gridCell row="COy-q6-8Qe" column="7OC-Fv-eMr" id="06u-gv-b4n">
+                            <popUpButton key="contentView" verticalHuggingPriority="750" translatesAutoresizingMaskIntoConstraints="NO" id="B41-eW-cJt">
+                                <rect key="frame" x="198" y="162" width="96" height="25"/>
+                                <popUpButtonCell key="cell" type="push" title="bold" bezelStyle="rounded" alignment="left" lineBreakMode="truncatingTail" state="on" borderStyle="borderAndBezel" imageScaling="proportionallyDown" inset="2" selectedItem="Ab6-Jt-UI5" id="ULp-DF-zeA">
+                                    <behavior key="behavior" lightByBackground="YES" lightByGray="YES"/>
+                                    <font key="font" metaFont="menu"/>
+                                    <menu key="menu" id="Wdx-5c-0xV">
+                                        <items>
+                                            <menuItem title="bold" state="on" id="Ab6-Jt-UI5"/>
```
Comment (p0deje) [inline Maccy/Settings/Base.lproj/AppearanceSettingsViewController.xib:351]: Let's use capitalized adjectives here:

* Bold
* Italic
* Underlined

File: Maccy/Settings/Base.lproj/AppearanceSettingsViewController.xib [hunk 6]
```diff
@@ -327,9 +330,38 @@ Default: 1500.</string>
                                 </customSpacing>
                             </stackView>
                         </gridCell>
+                        <gridCell row="COy-q6-8Qe" column="KLa-7P-lgK" id="sEN-zO-jeM">
+                            <textField key="contentView" focusRingType="none" horizontalHuggingPriority="251" verticalHuggingPriority="750" translatesAutoresizingMaskIntoConstraints="NO" id="I2c-rj-xuX" userLabel="Search string highlight:">
```
Comment (p0deje) [inline Maccy/Settings/Base.lproj/AppearanceSettingsViewController.xib:334]: What do you think about re-wording this to make it more concise, something like "Highlight search" or "Highlight match"?

**Review by p0deje — CHANGES_REQUESTED** [review]
First of all, thanks for the PR.

I've left a few minor comments, but most importantly, as I mentioned in the previous PR, there is no localization for other languages. I would need this to be able to merge it. Your best way to do so is to install the `bartycrouch` tool, run `bartycrouch update` in Terminal.app and update every localization file to with proper translation. I usually just use Google Translate for that.

File: Maccy/Menu/HistoryMenuItem/HistoryMenuItem.swift (hunk 2 above)
Comment (gongchoong) [reply Maccy/Menu/HistoryMenuItem/HistoryMenuItem.swift:28]: Done

File: Maccy/Settings/AppearanceSettingsViewController.swift (hunk 4 above)
Comment (gongchoong) [reply Maccy/Settings/AppearanceSettingsViewController.swift]: Done

File: Maccy/Settings/Base.lproj/AppearanceSettingsViewController.xib (hunk 5 above)
Comment (gongchoong) [reply Maccy/Settings/Base.lproj/AppearanceSettingsViewController.xib]: Done

File: Maccy/Settings/Base.lproj/AppearanceSettingsViewController.xib (hunk 6 above)
Comment (gongchoong) [reply Maccy/Settings/Base.lproj/AppearanceSettingsViewController.xib]: Highlight match makes sense. Done

**Comment by p0deje** [comment]
If you don't mind, I'll do some minor cleanup and merge it as a single commit.

**Comment by p0deje** [comment]
Actually, doesn't matter, I can just merge it as-is and fix it afterward. Thank you for contributing!

**Pull request merged**